pkg/segment/reader/record/rrcreader.go

```diff
@@ -396,12 +401,17 @@ func GetJsonFromAllRrc(allrrc []*utils.RecordResultContainer, esResponse bool, q
 		}
 	}
 
+	if aggs.HasGenerateEvent() {
+		processSingleSegment("generate_event", "generate_event", nil, true)
+	}
+
 	if !(tableColumnsExist || (aggs != nil && aggs.OutputTransforms == nil) || hasQueryAggergatorBlock || transactionArgsExist) {
 		allRecords, finalCols = applyHardcodedColumns(hardcodedArray, renameHardcodedColumns, allRecords, finalCols)
 		if len(hardcodedArray) > 0 {
 			numProcessedRecords = 1
 		}
 	} else {
+
```

Comment:
you should probably keep this in the below `if-else` block. 

I believe this PR might have some conflicts with my recent PR that got merged, there even with no `segmap`, in the else the function will call `processSegment`. So once you pull the latest changes you might understand.

Reply:
addressed in the latest commit.